src/Std/Data/HashMap/Raw.lean

```diff
@@ -0,0 +1,255 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Std.Data.DHashMap.Raw
+
+set_option linter.missingDocs true
+set_option autoImplicit false
+
+/-
+# Hash maps with unbundled well-formedness invariant
+
+This file develops the type `Std.Data.HashMap.Raw` of dependent hash
+maps with unbundled well-formedness invariant.
+
+This version is safe to use in nested inductive types. The well-formedness predicate is
+available as `Std.Data.HashMap.Raw.WF` and we prove in this file that all operations preserve
+well-formedness. When in doubt, prefer `HashMap` over `DHashMap.Raw`.
+
+Lemmas about the operations on `Std.Data.HashMap.Raw` are available in the module
+`Std.Data.HashMap.RawLemmas`.
+-/
+
+universe u v w
+
+variable {α : Type u} {β : Type v}
+
+namespace Std
+
+namespace HashMap
+
+/-- Hash maps without bundled well-formedness invariant. Suitable for use in nested
+inductive types. The well-formedness invariant is called `Raw.WF`. -/
```

Comment:
I think a note about the representation and use case would also be good here, in parallel with `DHashMap.Raw`.

Reply:
Already done :)